.github/workflows/cd-pr.yml

```diff
@@ -0,0 +1,37 @@
+name: Deploy PR-Preview
+
+on:
+  workflow_run:
+    workflows: ["Build PR-Preview"]
+    types:
+      - completed
+
+permissions:
```

Comment:
Once check if can remove any permission from here (as this workflow will have write permission

Reply:
we will need these permissions for deploying builds to github-pages. permissions:content: write

Reply:
pages: write
id-token: write

Are these necessary ?

Reply:
pages write gives permissions to deploy from github pages and id-token I don't know, it was there in build-lint so I used it

Reply:
Try removing permissions one by one and test if it is still working fine, let's allow as less permissions as we can

Reply:
done